### PR TITLE
Jw jan31

### PR DESCRIPTION
**Pull request by jeffwarnica**
Updates to the email processes, to class style, and, functionally, better detecting error states, as that somewhere broke. 

... and a bunch of empty commits because I can't git.

### CODE REVIEW

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb [hunk 1]
```diff
@@ -2,119 +2,123 @@
...
-  param_value ||= $evm.get_state_var(param.to_sym)
-  param_value ||= $evm.get_state_var(param.to_s)
-
-  $evm.log(:info, "{ '#{param}' => '#{param_value}' }") if @DEBUG
-  return param_value
-end
 
 # Determine the CFME/ManageIQ hostname
 #
 # @return CFME/ManageIQ hostname
```
Comment (itewk) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb]: fix indentation of comments.

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb [hunk 2]
```diff
@@ -2,119 +2,123 @@
...
+
+              log(:info, "from_email_address => #{from_email_address}") if @DEBUG
+              return from_email_address
+            end
 
 # Determin the email addresses to send the email to
 #
 # @param prov miq_provision to determine the to email addresses for
 #
 # @return array of email addresses to send the email to
```
Comment (itewk) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb]: fix indentation of comments.

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb [hunk 3]
```diff
@@ -142,127 +146,94 @@ def determine_to_email_addresses(prov)
 #  |----------------------------------------------------|---------------------------|
 #
 # -----------
```
Comment (itewk) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Provisioning/Email.class/__methods__/miqprovision_update.rb]: fix indentation of comments

**Comment by jeffwarnica** [comment]
Updated with requests, with bonus dump_root (disabled)

**Pull request merged**